src/data/markdown/docs/05 Examples/01 Examples/24 distribute-workloads.md

```diff
@@ -0,0 +1,229 @@
+---
+title: Distribute workloads across VUs
+excerpt: How to configure k6 so that some VUs do one action, and other VUs do another action
+---
+
+You can configure your k6 test scripts so that different sets of VUs execute different functions.
+This helps simulate traffic to real-world applications, where user behavior is rarely uniform.
+t
+For example, the majority of traffic to an e-commerce site might come from readers who only search items and read reviews. A small percentage might actively shop, performing actions that involve writes to the database and calls to different APIs.
+
+To account for these different behaviors, testers often want to distribute their VUs into different logical behaviors.
+k6 supports multiple ways to do this.
+The following sections provide different ways to split logic across VUs, along with some strategies to granularly filter results.
+
+<Blockquote mod="note" title="Aim for simplicity">
+
+These techniques could be used to create very complex configurations.
+However, more complexity creates more ambiguity for results interpretation
+
+</Blockquote>
+
+
+## Split logic across scenarios
+
+<Blockquote mod="note" title="">
+
+In this context, _workload_ refers to the traffic pattern simulated by a scenario.
+
+</Blockquote>
+
+
+One way to distribute traffic is to use scenarios to schedule different workloads for different functions.
+1. Define multiple scenarios in your `options.scenarios` object.
+1. Use the scenario `exec` property to execute different VU functions with a workload.
+
+For example, imagine a social media site that typically receives 100 concurrent users.
+Of those, 80 might visit their contacts page, and 20 might view the news.
+To configure such a distribution, make two scenarios with different throughput or VUs:
+
+```javascript
+import http from "k6/http";
+
+export const options = {
+  //scenario to view contacts
+  scenarios: {
+    contacts: {
+      executor: "shared-iterations",
+      exec: "contacts",
+      vus: 80,
+      iterations: 100,
+    },
+    //scenario to view news
+    news: {
+      executor: "shared-iterations",
+      exec: "news",
+      vus: 20,
+      iterations: 100,
+    },
+  },
+};
+
+//use the exec property to run different scenarios for different functions
+
+export function contacts() {
+  http.get("https://test.k6.io/contacts.php");
+}
+
+export function news() {
+  http.get("https://test.k6.io/news.php");
+}
+```
+
+### Tip: reduce scenario boilerplate with `new`
```

Comment:
For simplicity and less info overhead, I'd consider removing this section with the Tip.

Reply:
I also think this will probably be better off in the scenario docs or at least to be collapsed by default.

Also related to https://github.com/grafana/k6/issues/974